api/v1alpha1/amaltheasession_types.go

```diff
@@ -256,16 +255,37 @@ type Authentication struct {
 	// For `oauth2proxy` please see https://oauth2-proxy.github.io/oauth2-proxy/configuration/overview#config-file.
 	// Note that the `upstream` and `http_address` configuration options cannot be set from the secret because
 	// the operator knows how to set these options to the proper values.
-	SecretRef SessionSecretRef `json:"secretRef"`
+	SecretRef SessionSecretKeyRef `json:"secretRef"`
 	// +optional
 	// Additional volume mounts for the authentication container.
 	ExtraVolumeMounts []v1.VolumeMount `json:"extraVolumeMounts,omitempty"`
 }
 
 // A reference to a Kubernetes secret and a specific field in the secret to be used in a session
-type SessionSecretRef struct {
+type SessionSecretKeyRef struct {
 	Name string `json:"name"`
 	Key  string `json:"key"`
+	// +optional
+	// +kubebuilder:validation:Optional
+	// If the secret is adopted then the operator will delete the secret when the custom resource that uses it is deleted.
+	Adopt bool `json:"adopt"`
+}
+
+func (s *SessionSecretKeyRef) isAdopted() bool {
+	return s != nil && s.Name != "" && s.Adopt
+}
+
+// A reference to a whole Kubernetes secret where the key is not important
+type SessionSecretRef struct {
+	Name string `json:"name"`
+	// +optional
+	// +kubebuilder:validation:Optional
+	// If the secret is adopted then the operator will delete the secret when the custom resource that uses it is deleted.
+	Adopt bool `json:"adopt"`
+}
+
+func (s *SessionSecretRef) isAdopted() bool {
+	return s != nil && s.Name != "" && s.Adopt
```

Comment:
I did not expect that to be a possible implementation.

Thanks for the trick !